ide/api.lsp/src/org/netbeans/spi/lsp/CompletionCollector.java

```diff
@@ -0,0 +1,250 @@
+         */
+        @NonNull
+        public Builder label(@NonNull String label) {
+            this.label = label;
+            return this;
+        }
+
+        /**
+         * The kind of this completion.
+         */
+        @NonNull
+        public Builder kind(@NonNull Completion.Kind kind) {
+            this.kind = kind;
+            return this;
+        }
+
+        /**
+         * Adds tag for this completion.
+         */
+        @NonNull
+        public Builder addTag(@NonNull Completion.Tag tag) {
+            if (this.tags == null) {
+                this.tags = new ArrayList<>();
+            }
+            this.tags.add(tag);
+            return this;
+        }
+
+        /**
+         * A human-readable string with additional information
+         * about this completion, like type or symbol information.
+         */
+        @NonNull
+        public Builder detail(@NonNull CompletableFuture<String> detail) {
+            this.detail = detail;
+            return this;
+        }
+
+        /**
+         * A human-readable string that represents a doc-comment. An HTML format
+         * is supported.
+         */
+        @NonNull
+        public Builder documentation(@NonNull CompletableFuture<String> documentation) {
+            this.documentation = documentation;
+            return this;
+        }
+
+        /**
+         * Select this completion when showing.
+         */
+        @NonNull
+        public Builder preselect(boolean preselect) {
+            this.preselect = preselect;
+            return this;
+        }
+
+        /**
+         * A string that should be used when comparing this completion with other
+         * completions. When {@code null} the label is used as the sort text.
+         */
+        @NonNull
+        public Builder sortText(@NonNull String sortText) {
+            this.sortText = sortText;
+            return this;
+        }
+
+        /**
+         * A string that should be used when filtering a set of completions.
+         * When {@code null} the label is used as the filter.
+         */
+        @NonNull
+        public Builder filterText(@NonNull String filterText) {
+            this.filterText = filterText;
+            return this;
+        }
+
+        /**
+         * A string that should be inserted into a document when selecting
+         * this completion. When {@code null} the label is used as the insert text.
+         */
+        @NonNull
+        public Builder insertText(@NonNull String insertText) {
+            this.insertText = insertText;
+            return this;
+        }
+
+        /**
+         * The format of the insert text. The format applies to both the
+         * {@code insertText} property and the {@code newText} property of a provided
+         * {@code textEdit}. If omitted defaults to {@link TextFormat#PlainText}.
+         */
+        @NonNull
+        public Builder insertTextFormat(@NonNull Completion.TextFormat insertTextFormat) {
```

Comment:
I suggest to join the methods `insertText` and its format into one with two parameters.

Reply:
The `insertTextFormat` applies to both the `insertText` property and the `newText` property of a provided `textEdit`. I would leave them separated.